app/client/lib/MultiUserManager.ts

```diff
@@ -0,0 +1,180 @@
+import {computed, Computed, dom, DomElementArg, IDisposableOwner, Observable, styled} from "grainjs";
+import {cssModalBody, cssModalButtons, cssModalTitle, IModalControl, modal, cssAnimatedModal} from 'app/client/ui2018/modals';
+import {bigBasicButton, bigPrimaryButton} from 'app/client/ui2018/buttons';
+import {mediaXSmall, testId, theme, vars} from 'app/client/ui2018/cssVars';
+import {UserManagerModel, IOrgMemberSelectOption} from 'app/client/models/UserManagerModel';
+import {icon} from 'app/client/ui2018/icons';
+import {textarea} from "app/client/ui/inputs";
+import {BasicRole, VIEWER, NonGuestRole, isBasicRole} from "app/common/roles";
+import {menu, menuItem} from 'app/client/ui2018/menus';
+
+function parseEmailList(emailListRaw: string): Array<string> {
+  return emailListRaw
+    .split('\n')
+    .map(email => email.trim().toLowerCase())
+    .filter(email => email !== "")
+}
+
+function validateEmail(email: string): boolean {
+  const mailformat = /\S+@\S+\.\S+/;
+  return mailformat.test(email);
+}
+
+export function buildMultiUserManagerModal(
+  owner: IDisposableOwner,
+  model: UserManagerModel,
+  onAdd: (email: string, role: NonGuestRole) => void,
+) {
+  const emailListObs = Observable.create(owner, "");
+  const rolesObs = Observable.create<BasicRole>(owner, VIEWER);
+  const isValidObs = Observable.create(owner, true);
+  
+  const enableAdd: Computed<boolean> = computed((use) => Boolean(use(emailListObs) && use(rolesObs) && use(isValidObs)));
+  
+  const save = (ctl: IModalControl) => {
+    const emailList = parseEmailList(emailListObs.get())
+    const role = rolesObs.get();
+    if (emailList.some(email => !validateEmail(email))) {
+      isValidObs.set(false);
+    } else {
+      emailList.map(email => onAdd(email, role))
+      ctl.close()
+    }
+  }
+
+  return modal(ctl => [
+    // We set the padding to 0 since the body scroll shadows extend to the edge of the modal.
+    { style: 'padding: 0;' },
+    dom.cls(cssAnimatedModal.className),
+    cssTitle(
+      'Invite Users',
+      testId('um-header'),
+    ),
+    cssModalBody(
+      cssUserManagerBody(
+        buildEmailsTextarea(emailListObs, isValidObs),
+        dom.domComputed(isValidObs, isValid => !isValid ? cssErrorMessage('At least one email is invalid') : null),
```

Comment:
```suggestion
        dom.maybe(use => !use(isValidObs), () => cssErrorMessage('At least one email is invalid')),
```

A handy shortcut for when you want to conditionally render DOM elements.